src/library/scala/collection/generic/IsMapLike.scala

```diff
@@ -0,0 +1,66 @@
+package scala.collection
+package generic
+
+import IsMapLike.Tupled
+
+/**
+  * Type class witnessing that a collection type `C`
+  * has keys of type `K`, values of type `V` and has a conversion to `MapOps[K, V, _, _]`.
+  *
+  * This type enables simple enrichment of `Map`s with extension methods.
+  *
+  * @tparam Repr Collection type (e.g. `Map[Int, String]`)
+  */
+trait IsMapLike[Repr] extends IsIterableLike[Repr] {
+
+  /** The type of keys */
+  type K
+
+  /** The type of values */
+  type V
+
+  type A = (K, V)
+
+  /** A conversion from the type `C` to `MapOps[K, V, _, _]` */
+  override def apply(c: Repr): MapOps[K, V, Tupled[Iterable]#Ap, C]
+
+}
+
+object IsMapLike extends LowPriorityIsMapLike {
+
+  /** Convenient type level function that takes a unary type constructor `F[_]`
+    * and returns a binary type constructor that tuples its parameters and passes
+    * them to `F`.
+    *
+    * `Tupled[F]#Ap` is equivalent to `({ type Ap[X, +Y] = F[(X, Y)] })#Ap`.
+    */
+  type Tupled[F[+_]] = { type Ap[X, Y] = F[(X, Y)] }
+
+  // 1. Map collections
+  implicit def mapOpsIsMapLike[CC[X, Y] <: MapOps[X, Y, Tupled[Iterable]#Ap, CC[X, Y]], K0, V0]: IsMapLike[CC[K0, V0]] { type K = K0; type V = V0; type C = CC[K, V] } =
+    new IsMapLike[CC[K0, V0]] {
+      type K = K0
+      type V = V0
+      type C = CC[K0, V0]
+      def apply(c: CC[K0, V0]): MapOps[K0, V0, Tupled[Iterable]#Ap, C] = c
+    }
+
+  // 2. MapView
+  implicit def mapViewIsMapLike[CC[X, Y] <: MapView[X, Y], K0, V0]: IsMapLike[CC[K0, V0]] { type K = K0; type V = V0; type C = View[(K, V)] } =
```

Comment:
Why is `MapView` treated specially? `MapView` extends `MapOps`.

Reply:
This is the same reason as with `SortedMap` above.